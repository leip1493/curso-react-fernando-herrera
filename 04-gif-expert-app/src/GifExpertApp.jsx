import React, { useState } from "react";
import PropTypes from 'prop-types';
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GidExpertApp</h2>

      <AddCategory {...{ setCategories }} />

      <hr />

      <ol>
        {categories.map(category => (
          <GifGrid key={category} {...{ category }} />
        ))}
      </ol>
    </>
  );
};

GifExpertApp.propTypes = {
  defaultCategories: PropTypes.arrayOf(PropTypes.string),
}

GifExpertApp.defaultProps = {
  defaultCategories: ["Naruto"]
};
