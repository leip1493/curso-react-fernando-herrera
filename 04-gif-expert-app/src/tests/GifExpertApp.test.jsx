import React from "react";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("<GifExpertApp />", () => {
  test("Debe renderizar exitosamente ", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar una lista de categorias de gifs ", () => {
    const categories = ["DBZ", "ONE PIECE"];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
