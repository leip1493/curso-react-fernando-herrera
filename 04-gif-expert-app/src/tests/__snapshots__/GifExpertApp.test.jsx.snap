// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> Debe mostrar una lista de categorias de gifs  1`] = `
<Fragment>
  <h2>
    GidExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="DBZ"
      key="DBZ"
    />
    <GifGrid
      category="ONE PIECE"
      key="ONE PIECE"
    />
  </ol>
</Fragment>
`;

exports[`<GifExpertApp /> Debe renderizar exitosamente  1`] = `
<Fragment>
  <h2>
    GidExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;
